### ISSUE TITLE
Interpreting revision labels as YANG semantic version numbers (aka versioning scheme extension)

### ISSUE THREAD

**Issue by Reshad-Rahman** [netmod-wg/yang-ver-dt#48]
From Martin Bjorklund on draft-ietf-netmod-yang-module-versioning-00:

o  3.3

    All revision labels that match the pattern for the "version"
    typedef in the ietf-yang-semver YANG module MUST be interpreted as
    YANG semantic version numbers.

  I don't think this is a good idea.  Seems like a layer violation.
  What if my project use another dialect of semver, that wouldn't be
  possible with this rule.  I think this needs to be removed.

**Comment by Reshad-Rahman** [comment]
https://mailarchive.ietf.org/arch/browse/netmod/?q=Interpreting%20revision%20labels%20as%20YANG%20semantic%20version%20numbers

At the virtual interim (April 2nd) there seemed to be consensus to have another statement to declare what versioning scheme is being used. 

At the authors meeting on April 14th, we discussed 
1) 1 extension which takes in as parameter the scheme. Do identity for versioning schemes? No canonical format for identity.
2) 1 extension per versioning scheme.
There seems to be preference for 2. If we go that route, we should consider having a guideline how those extensions should be named. e.g. versiong-scheme-xxx. That guideline should go in module versioning draft and semver draft will define its extension.

**Comment by Reshad-Rahman** [comment]
April 21st: only 1 revision-label scheme is simpler. Double-check with Balazs if this is ok.

**Comment by jsterne** [comment]
Note - this is related to issue #18

**Comment by Reshad-Rahman** [comment]
We discussed whether the versioning-scheme should be top-level or per revision-label statement. Consensus is that top-level, i.e. per-module, is ok and that per revision-label statement seems to be overkill right now.

We also discussed whether the versioning-scheme is allowed to change during the lifetime of a module. The consensus is yes. e.g. a module may go from yang-semver to vendor-xxx or vice-versa.

Last discussion point is whether we should remove the old revision-labels (belonging to the old scheme) when a new versioning scheme is used. Text will be modified that the old revision-labels must be removed/replaced (as per text suggestion from Rob)
